server/routes/api/representatives.js

```diff
@@ -50,15 +50,32 @@ const ALLOWED_JURISDICTION_FILTERS = Object.keys(JURISDICTION_FILTER_MAP)
 // Endpoints
 
 // Get
-router.get('/:zipCode', async (req, res) => {
-  const { zipCode } = req.params
+router.get('/:searchText', async (req, res) => {
+  const { searchText } = req.params
   const { filter } = req.query
 
-  if (!zipCode.match(/^\d{5}(-\d{4})?$/)) {
+  /* check if valid ZIP code 
+    ^\d{5} - the start of the line has 5 digits
+    (-\d{4})?$ the end of the line has 4 digits preceded by a dash (optionally)
+  */
+  let isValidZIPcode = /^\d{5}(-\d{4})?$/.test(searchText)
+  /* check if valid street address
+     ^(\d{3,})\s? the start of the line can have 3 or more digits followed by a space
+     (\w{0,})\s the next part can have 2 or more letters followed by a space
```

Comment:
This part will not match addresses like `123 45th street` - Addresses with ordinal parts (e.g. 1st, 2nd, 3rd) would be invalid.  I would recommend something similar to `(\d+\w{2})|((\w{2,})\s?)+` (I didn't test this) to match ordinal addresses and also street names with multiple words

Reply:
Hello, this part will not accept a string like 123 45th street, because it expects 2 or more alphabetical characters for the second group. Addresses with ordinal parts (e.g., 1st, 2nd, 3rd) would be valid if they had a third group with additional alphabetical characters.  The w{start, finish} already implies any word character (letter, number, underscore). That is why I somehow feel (\d+\w{2})|((\w{2,})\s?)+ could be redundant as we are already implying we can accept any word character (numbers or alphabetical characters as part of the second group). I have updated the third group to be optional and take strings that do not have two additional words after the second group, let me know what you think!

Reply:
@thyeggman 
**The update:**
`(\d{3,})\s?(\w{0,5})\s([a-zA-Z]{2,30})\s?([a-zA-Z]{2,15})\.?\s?(\w{0,5})$`